File: packages/shared/src/enums/ImageSize.ts
// Valid image widths for generating CDN url.
export enum ImageSize {
  W_24 = 24,
  W_32 = 32,
  W_64 = 64,
  W_128 = 128,
  W_256 = 256,
  W_320 = 320,
  W_640 = 640,
  W_860 = 860,
  W_1024 = 1024,
  W_1440 = 1440,
  W_1920 = 1920,
  W_2048 = 2048,
}

export type TImageSize = `${Extract<
  ImageSize,
  number
>}` extends `${infer N extends number}`
  ? N
  : never;

File: apps/backend/src/constants/image_size.rs
use std::fmt;

/// Valid image widths for generating CDN URLs.
#[derive(Debug, Copy, Clone)]
pub enum ImageSize {
    W64 = 64,
    W128 = 128,
    W320 = 320,
    W640 = 640,
    W960 = 960,
    W1200 = 1200,
    W1440 = 1440,
    W1920 = 1920,
    W2440 = 2440,
}

impl fmt::Display for ImageSize {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        let value = *self as u32;
        fmt::Debug::fmt(&value, f)
    }
}
